tests/Scanner.test.ts

```diff
@@ -0,0 +1,39 @@
+import { continue_lines, scan } from '../src/Query/Scanner';
+
+describe('continue_lines', () => {
+    it('keeps non-continued text the same', () => {
+        const text = 'not done\ndue this week';
+        expect(continue_lines(text)).toEqual(text);
+    });
+    it('removes backslashed newlines', () => {
+        const text = 'line1 \\\ncontinued';
+        expect(continue_lines(text)).toEqual('line1 continued');
+    });
+    it('collapses escaped final backslashes', () => {
+        const text = 'line1 \\\\\nline2';
+        expect(continue_lines(text)).toEqual('line1 \\\nline2');
+    });
+    it('preserves non-final backslashes', () => {
+        const text = 'line\\1 \\\ncontinued \\\\\\\nline2';
+        expect(continue_lines(text)).toEqual('line\\1 continued \\\\\nline2');
```

Comment:
I'm finding these tests quite hard to picture in my mind...

It really helps for tests to not just be correct, but also to be clear.

Is there some kind of raw string that you could  use to avoid the double backslashes...

And could the '\n' be replaced by actual newlines inside the strings?

---

I experimented with safely, incrementally, changing this test:

from this:

```typescript
    it('preserves non-final backslashes', () => {
        const text = 'line\\1 \\\ncontinued \\\\\\\nline2';
        expect(continue_lines(text)).toEqual('line\\1 continued \\\\\nline2');
    });
```

to this:

```typescript
    it('preserves non-final backslashes', () => {
        const text = String.raw`line\1 \
continued \\\
line2`;
        expect(continue_lines(text)).toEqual(String.raw`line\1 continued \\
line2`);
    });
```

In order to make the change carefully, I did the following separate steps (from memory), checking that the tests passed at each stage:

1. On `text`:
    1. Change quotes from ' to '
    2. Change one `\n` to an actual newline character, and removed the indentation that the IDE added
    3. Changed other `\n` to new line
    4. Added `String.raw` - tests failed then
    5. Converted the double-slashes to single-slashes
2. On the `toEqual()` string
    1. Repeat the same steps...

By keeping the 'expected' the same whilst editing the input, it gives me more confidence that I have not changed the behaviour of the test.

Reply:
Actually the above steps are now not “from memory” as I redid the edits to make sure what I was saying was correct.

Reply:
I will change the string style. It's unfortunate that multi-line strings have to be flush left.

Reply:
Fixed.